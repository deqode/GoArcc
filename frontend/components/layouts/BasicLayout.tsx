import { Grid, makeStyles, Paper, Typography } from '@material-ui/core'
import React, { ReactElement } from 'react'

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: 100,
    paddingBottom: 30,
    flexGrow: 1,
  },
  paper: {
    padding: theme.spacing(2),
    textAlign: 'center',
  },
  text: {
    display: 'inline-block',
  },
}))
const BasicLayout = ({
  heading,
  subHeading,
  component,
}: {
  heading: string
  subHeading?: string
  component: ReactElement | string | string[]
}): ReactElement => {
  const classes = useStyles()
  return (
    <Grid
      container
      direction="row"
      justify="space-around"
      alignItems="center"
      className={classes.root}>
      <Grid item xs={12} sm={6} md={4} lg={4} xl={4}>
        <Paper className={classes.paper} elevation={0}>
          <Typography color="primary" variant="h3" component="h1">
            {heading}
          </Typography>
          <br />
          <Typography color="secondary" variant="h5">
            {subHeading}
          </Typography>
        </Paper>
      </Grid>
      <Grid item xs={12} sm={6} md={4} lg={4} xl={4}>
        <Paper className={classes.paper} elevation={0}>
          <Typography display="inline" className={classes.text}>
            {component}
          </Typography>
        </Paper>
      </Grid>
    </Grid>
  )
}

export default BasicLayout
