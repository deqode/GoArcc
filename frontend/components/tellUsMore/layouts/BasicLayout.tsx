import { Grid, makeStyles } from '@material-ui/core'
import React, { ReactElement } from 'react'
const useStyles = makeStyles({
  root: {
    paddingTop: 100,
  },
})
const BasicLayout = ({
  heading,
  component,
}: {
  heading: ReactElement | string
  component: ReactElement | string
}): ReactElement => {
  const classes = useStyles()
  return (
    <Grid container className={classes.root}>
      <Grid item xs={6} sm={2} md={8} lg={12} xl={12}>
        {heading}
      </Grid>
      <Grid item xs={6} sm={10} md={4} lg={12} xl={12}>
        {component}
      </Grid>
    </Grid>
  )
}

export default BasicLayout
