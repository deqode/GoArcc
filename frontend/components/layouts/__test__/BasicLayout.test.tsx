import { Button } from '@material-ui/core'
import renderer from 'react-test-renderer'

import BasicLayout from '../BasicLayout'

describe('basic layout component tests', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <BasicLayout
          heading="This is basicLayout headding"
          subHeading="This is basicLayout sub-headding"
          component={
            <Button variant="contained" color="primary" href={'https://github.com/'}>
              SignUP
            </Button>
          }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
