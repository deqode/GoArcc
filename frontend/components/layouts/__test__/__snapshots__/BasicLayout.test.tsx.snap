// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Basic Layout  snapshot should match the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
  >
    <div
      className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <h1
        className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
      >
        This is basicLayout headding
      </h1>
      <br />
      <h5
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      >
        This is basicLayout sub-headding
      </h5>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
  >
    <div
      className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <p
        className="MuiTypography-root makeStyles-text-3 MuiTypography-body1 MuiTypography-displayInline"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          href="https://github.com/"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            SignUP
          </span>
        </a>
      </p>
    </div>
  </div>
</div>
`;
